syntax = "proto3";

service ClientConnection
{
    rpc Connect(stream ClientMessage) returns (stream ServerMessage);
}

message ServerMessage
{
    TrainOrder trainOrder = 1;
    EvalOrder evalOrder = 2;
    DisconnectOrder disconnectOrder = 3;
    SetParamsOrder setParamsOrder = 4;
}

message ClientMessage
{
    TrainResponse trainResponse = 1;
    EvalResponse evalResponse = 2;
    SetParamsResponse setParamsResponse = 3;
}

message TrainOrder
{
    bytes modelParameters = 1;
    bytes configDict = 2;
}

message TrainResponse
{
    bytes modelParameters = 1;
    bytes responseDict = 2;
}

message EvalOrder
{
    bytes modelParameters = 1;
    bytes configDict = 2;
}

message EvalResponse
{
    bytes responseDict = 1;
}

message SetParamsOrder
{
    bytes modelParameters = 1;
}

message SetParamsResponse
{

}

message DisconnectOrder
{
    string message = 1;
    int32 reconnectTime = 2;
}